### PR TITLE
raftstore: add detail log for large region

### PR DESCRIPTION
**Pull request by Connor1996**
### What problem does this PR solve?

Add detailed log for large regions with printing the ssts involved and corresponding size/keys to give more information to investigate related problems.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

- No code

### Release note

### CODE REVIEW

File: components/raftstore/src/coprocessor/split_check/keys.rs [hunk 1]
```diff
@@ -199,14 +200,42 @@ pub fn get_region_approximate_keys_cf(
...
+                        .file_name()
+                        .map(|f| f.to_str().unwrap())
+                        .unwrap_or(&*k),
+                    key
+                )
+            })
+            .collect::<Vec<_>>()
+            .join(", ");
+        info!(
+            "region key is too many";
```
Comment (BusyJay) [inline components/raftstore/src/coprocessor/split_check/keys.rs]:
```suggestion
            "region contains too many keys";
```

File: components/raftstore/src/coprocessor/split_check/keys.rs [hunk 2]
```diff
@@ -199,14 +200,42 @@ pub fn get_region_approximate_keys_cf(
...
-    Ok(keys)
+
+    if total_keys > 10000000 {
+        let ssts = collection
+            .iter()
+            .map(|(k, v)| {
+                let props = RangeProperties::decode(&v.user_collected_properties()).unwrap();
+                let key = props.get_approximate_keys_in_range(&start_key, &end_key);
+                format!(
+                    "{} with key {}",
```
Comment (BusyJay) [inline components/raftstore/src/coprocessor/split_check/keys.rs]: If there are a lot of ssts, the output can be too verbose.

File: components/raftstore/src/coprocessor/split_check/keys.rs [hunk 3]
```diff
@@ -199,14 +200,42 @@ pub fn get_region_approximate_keys_cf(
...
 
     let collection = box_try!(db.get_range_properties_cf(cfname, &start_key, &end_key));
     for (_, v) in collection.iter() {
         let props = box_try!(RangeProperties::decode(&v.user_collected_properties()));
-        keys += props.get_approximate_keys_in_range(&start_key, &end_key);
+        total_keys += props.get_approximate_keys_in_range(&start_key, &end_key);
     }
-    Ok(keys)
+
+    if total_keys > 10000000 {
```
Comment (siddontang) [inline components/raftstore/src/coprocessor/split_check/keys.rs]: em, don't use magic number directly here.

File: components/raftstore/src/coprocessor/split_check/keys.rs [hunk 4]
```diff
@@ -195,18 +204,47 @@ pub fn get_region_approximate_keys_cf(
...
 
     let collection = box_try!(db.get_range_properties_cf(cfname, &start_key, &end_key));
     for (_, v) in collection.iter() {
         let props = box_try!(RangeProperties::decode(&v.user_collected_properties()));
-        keys += props.get_approximate_keys_in_range(&start_key, &end_key);
+        total_keys += props.get_approximate_keys_in_range(&start_key, &end_key);
     }
-    Ok(keys)
+
+    if total_keys > large_threshold && large_threshold != 0 {
```
Comment (zhangjinpeng1987) [inline components/raftstore/src/coprocessor/split_check/keys.rs]: `if large_threshold != 0 && total_keys > large_threshold`

File: components/raftstore/src/coprocessor/split_check/keys.rs [hunk 5]
```diff
@@ -195,18 +204,47 @@ pub fn get_region_approximate_keys_cf(
...
+        let ssts = collection
+            .iter()
+            .map(|(k, v)| {
+                let props = RangeProperties::decode(&v.user_collected_properties()).unwrap();
+                let key = props.get_approximate_keys_in_range(&start_key, &end_key);
+                format!(
+                    "({} {})",
+                    Path::new(&*k)
+                        .file_name()
```
Comment (zhangjinpeng1987) [inline components/raftstore/src/coprocessor/split_check/keys.rs]: I guess this log will annoying, maybe we just need to know how many ssts the properties got from.

File: components/raftstore/src/coprocessor/split_check/keys.rs [hunk 6]
```diff
@@ -195,18 +204,47 @@ pub fn get_region_approximate_keys_cf(
...
-    Ok(keys)
+
+    if total_keys > large_threshold && large_threshold != 0 {
+        let ssts = collection
+            .iter()
+            .map(|(k, v)| {
+                let props = RangeProperties::decode(&v.user_collected_properties()).unwrap();
+                let key = props.get_approximate_keys_in_range(&start_key, &end_key);
+                format!(
+                    "({} {})",
```
Comment (zhangjinpeng1987) [inline components/raftstore/src/coprocessor/split_check/keys.rs]: Maybe print sst count is enough

File: components/raftstore/src/coprocessor/split_check/keys.rs (hunk 4 above)
Comment (Connor1996) [reply components/raftstore/src/coprocessor/split_check/keys.rs]: mostly,  large_threshold != 0

File: components/raftstore/src/coprocessor/split_check/keys.rs (hunk 5 above)
Comment (Connor1996) [reply components/raftstore/src/coprocessor/split_check/keys.rs]: what if the size from one sst is abnormal, we can't know which sst is

File: components/raftstore/src/coprocessor/split_check/keys.rs (hunk 4 above)
Comment (zhangjinpeng1987) [reply components/raftstore/src/coprocessor/split_check/keys.rs]: Please reading code like reading an article.

File: components/raftstore/src/coprocessor/split_check/keys.rs [hunk 7]
```diff
@@ -195,18 +204,47 @@ pub fn get_region_approximate_keys_cf(
...
+        total_keys += props.get_approximate_keys_in_range(&start_key, &end_key);
     }
-    Ok(keys)
+
+    if total_keys > large_threshold && large_threshold != 0 {
+        let ssts = collection
+            .iter()
+            .map(|(k, v)| {
+                let props = RangeProperties::decode(&v.user_collected_properties()).unwrap();
+                let key = props.get_approximate_keys_in_range(&start_key, &end_key);
```
Comment (zhangjinpeng1987) [inline components/raftstore/src/coprocessor/split_check/keys.rs]: s/key/keys

File: components/raftstore/src/coprocessor/split_check/keys.rs (hunk 6 above)
Comment (zhangjinpeng1987) [reply components/raftstore/src/coprocessor/split_check/keys.rs]: I think the log will be `ssts [1.sst 100, 2.sst 100, 3.sst 150 ...]`, I guess users cannot get what is 100 and 150 mean.

**Comment by Connor1996** [comment]
PTAL

**Pull request merged**